File: app/src/main/java/com/yeraygarcia/recipes/util/NetworkUtil.java
package com.yeraygarcia.recipes.util;

import android.content.Context;
import android.net.ConnectivityManager;

public class NetworkUtil {

    private NetworkUtil() {
        // prevent instantiation
    }

    public static boolean isOnline(Context context) {
        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        return cm != null && cm.getActiveNetworkInfo() != null && cm.getActiveNetworkInfo().isConnectedOrConnecting();
    }
}

File: app/src/main/java/com/yeraygarcia/recipes/util/NetworkUtil.kt
package com.yeraygarcia.recipes.util

import android.content.Context
import android.net.ConnectivityManager

object NetworkUtil {

    fun isOnline(context: Context): Boolean {
        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        return cm.activeNetworkInfo?.isConnectedOrConnecting ?: false
    }
}// prevent instantiation
